--- a/src/testutils/CMakeLists.txt	2024-12-14 19:58:44.000000000 +0800
+++ b/src/testutils/CMakeLists.txt	2024-12-14 20:00:39.000000000 +0800
@@ -63,7 +63,7 @@
     check_hip_path()
     get_property(HIP_ADD_LIBRARY_FOUND GLOBAL PROPERTY GMX_HIP_ADD_LIBRARY_FOUND)
     if(NOT HIP_ADD_LIBRARY_FOUND)
-        list(APPEND CMAKE_MODULE_PATH ${ROCM_PATH}/hip/cmake)
+        list(APPEND CMAKE_MODULE_PATH ${ROCM_PATH}/lib/cmake/hip)
         # set(CMAKE_MODULE_PATH "${ROCM_PATH}/cmake" ${CMAKE_MODULE_PATH})
         find_package(HIP)
 	    set_property(GLOBAL PROPERTY GMX_HIP_ADD_LIBRARY_FOUND true)
