diff --git a/src/lib_proj.cmake b/src/lib_proj.cmake
index a35eeb8c..b98a435d 100644
--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -470,7 +470,7 @@ endif()
 
 if(CURL_ENABLED)
   target_compile_definitions(proj PRIVATE -DCURL_ENABLED)
-  target_include_directories(proj PRIVATE ${CURL_INCLUDE_DIR})
+  target_include_directories(proj PRIVATE ${CURL_INCLUDE_DIRS})
   target_link_libraries(proj
     PRIVATE
       ${CURL_LIBRARY}
diff --git a/test/unit/CMakeLists.txt b/test/unit/CMakeLists.txt
index eb41a6c8..a09bed62 100644
--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -144,7 +144,7 @@ add_executable(test_network
   main.cpp
   test_network.cpp)
 if(CURL_ENABLED)
-  include_directories(${CURL_INCLUDE_DIR})
+  include_directories(${CURL_INCLUDE_DIRS})
   target_compile_definitions(test_network PRIVATE -DCURL_ENABLED)
   target_link_libraries(test_network PRIVATE ${CURL_LIBRARY})
 endif()
