diff --git a/src/lib_proj.cmake b/src/lib_proj.cmake
index f2de1c10..034b601d 100644
--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -413,7 +413,7 @@ if(CURL_ENABLED)
   target_include_directories(proj PRIVATE ${CURL_INCLUDE_DIR})
   target_link_libraries(proj
     PRIVATE
-      ${CURL_LIBRARY}
+      ${CURL_LIBRARIES}
       $<$<CXX_COMPILER_ID:MSVC>:ws2_32>
       $<$<CXX_COMPILER_ID:MSVC>:wldap32>
       $<$<CXX_COMPILER_ID:MSVC>:advapi32>
