--- a/src/arch/ofi/machine.C
+++ b/src/arch/ofi/machine.C
@@ -85,6 +85,13 @@
 #include <rdma/fi_domain.h>
 #include <rdma/fi_tagged.h>
 #include <rdma/fi_rma.h>
+#include <stddef.h>
+
+#ifndef container_of
+#define container_of(ptr, type, member) \
+    ((type *)((char *)(ptr) - offsetof(type, member)))
+#endif
+
 
 #define USE_OFIREQUEST_CACHE 0
 
@@ -312,7 +319,7 @@
      *  - FI_MR_BASIC requires us to register the RMA buffers
      *    and to exchange the keys.
      */
-    enum fi_mr_mode mr_mode;
+    int mr_mode;
 
     /** Used as unique key value in FI_MR_SCALABLE mode */
     uint64_t mr_counter;
@@ -441,7 +448,7 @@
     OFI_INFO("use inject: %d\n", context.use_inject);
 
     context.rma_maxsize = prov->ep_attr->max_msg_size;
-    context.mr_mode = static_cast<fi_mr_mode>(prov->domain_attr->mr_mode);
+    context.mr_mode = prov->domain_attr->mr_mode;
     context.mr_counter = 0;
 
     OFI_INFO("maximum rma size: %ld\n", context.rma_maxsize);
