diff --git a/src/gromacs/modularsimulator/modularsimulator.h b/src/gromacs/modularsimulator/modularsimulator.h
--- a/src/gromacs/modularsimulator/modularsimulator.h
+++ b/src/gromacs/modularsimulator/modularsimulator.h
@@ -49,6 +49,8 @@
 #ifndef GROMACS_MODULARSIMULATOR_MODULARSIMULATOR_H
 #define GROMACS_MODULARSIMULATOR_MODULARSIMULATOR_H
 
+#include <cstdlib>
+
 #include "gromacs/mdrun/isimulator.h"
 
 struct CheckpointHeaderContents;
