diff --git a/symbol.h b/symbol.h
--- a/symbol.h
+++ b/symbol.h
@@ -98,10 +98,10 @@
 #define is_attrset_sym(sym) (sym_type(sym)==ID_ATTRSET)
 #define is_const_sym(sym) (sym_type(sym)==ID_CONST)
 #define is_class_sym(sym) (sym_type(sym)==ID_CLASS)
 #define is_junk_sym(sym) (sym_type(sym)==ID_JUNK)
 
-RUBY_FUNC_EXPORTED const uint_least32_t ruby_global_name_punct_bits[(0x7e - 0x20 + 31) / 32];
+extern const uint_least32_t ruby_global_name_punct_bits[(0x7e - 0x20 + 31) / 32];
 
 static inline int
 is_global_name_punct(const int c)
 {
