--- orig.singularity.lua	2022-06-08 10:22:12.000000000 +0800
+++ singularity.lua	2022-06-08 10:24:49.000000000 +0800
@@ -9,7 +9,7 @@
 This module is a singularity container wrapper for {{ name }} v{{ version }}
 {% if description %}{{ description }}{% endif %}
 
-Container:
+Container (available through variable SINGULARITY_CONTAINER):
 
  - {{ container_sif }}
 
@@ -25,6 +25,8 @@
        singularity inspect -r <container>
  - {|module_name|}-inspect-deffile:
        singularity inspect -d <container>
+ - {|module_name|}-container:
+       echo "$SINGULARITY_CONTAINER"
 
 {% if aliases %}{% for alias in aliases %} - {{ alias.name }}:
        singularity exec {% if features.gpu %}{{ features.gpu }} {% endif %}{% if features.home %}-B {{ features.home }} --home {{ features.home }} {% endif %}{% if features.x11 %}-B {{ features.x11 }} {% endif %}{% if settings.environment_file %}-B <moduleDir>/{{ settings.environment_file }}:/.singularity.d/env/{{ settings.environment_file }}{% endif %} {% if settings.bindpaths %}-B {{ settings.bindpaths }} {% endif %}{% if alias.singularity_options %}{{ alias.singularity_options }} {% endif %}<container> {{ alias.command }} "$@"
@@ -36,10 +38,20 @@
  - SINGULARITY_COMMAND_OPTS: to define custom options for the command (e.g., -b)
 ]]) 
 
+-- Enforce explicit usage of versions by requiring full module name
+if (mode() == "load") then
+  if (myModuleUsrName() ~= myModuleFullName()) then
+    LmodError(
+        "Default module versions are disabled by your systems administrator.\n\n",
+        "\tPlease load this module as <name>/<version>.\n"
+    )
+  end
+end
+
 {% if settings.singularity_module %}load("{{ settings.singularity_module }}"){% endif %}
 
--- directory containing this modulefile (dynamically defined)
-local moduleDir = myFileName():match("(.*[/])") or "."
+-- directory containing this modulefile, once symlinks resolved (dynamically defined)
+local moduleDir = subprocess("realpath " .. myFileName()):match("(.*[/])") or "."
 
 -- singularity environment variable to set shell
 setenv("SINGULARITY_SHELL", "{{ settings.singularity_shell }}")
@@ -48,8 +60,12 @@
 if not os.getenv("SINGULARITY_OPTS") then setenv ("SINGULARITY_OPTS", "") end
 if not os.getenv("SINGULARITY_COMMAND_OPTS") then setenv ("SINGULARITY_COMMAND_OPTS", "") end
 
--- interactive shell to any container, plus exec for aliases
 local containerPath = '{{ container_sif }}'
+-- service environment variable to access full SIF image path
+setenv("SINGULARITY_CONTAINER", containerPath)
+set_shell_function("{|module_name|}-container", "echo " .. containerPath)
+
+-- interactive shell to any container, plus exec for aliases
 local shellCmd = "singularity ${SINGULARITY_OPTS} shell ${SINGULARITY_COMMAND_OPTS} -s {{ settings.singularity_shell }} {% if features.gpu %}{{ features.gpu }} {% endif %}{% if features.home %}-B {{ features.home }} --home {{ features.home }} {% endif %}{% if features.x11 %}-B {{ features.x11 }} {% endif %}{% if settings.environment_file %}-B " .. moduleDir .. "/{{ settings.environment_file }}:/.singularity.d/env/{{ settings.environment_file }}{% endif %} {% if settings.bindpaths %}-B {{ settings.bindpaths }}{% endif %} " .. containerPath
 local execCmd = "singularity ${SINGULARITY_OPTS} exec ${SINGULARITY_COMMAND_OPTS} {% if features.gpu %}{{ features.gpu }} {% endif %}{% if features.home %}-B {{ features.home }} --home {{ features.home }} {% endif %}{% if features.x11 %}-B {{ features.x11 }} {% endif %}{% if settings.environment_file %}-B " .. moduleDir .. "/{{ settings.environment_file }}:/.singularity.d/env/{{ settings.environment_file }}{% endif %} {% if settings.bindpaths %}-B {{ settings.bindpaths }}{% endif %} "
 local runCmd = "singularity ${SINGULARITY_OPTS} run ${SINGULARITY_COMMAND_OPTS} {% if features.gpu %}{{ features.gpu }} {% endif %}{% if features.home %}-B {{ features.home }} --home {{ features.home }} {% endif %}{% if features.x11 %}-B {{ features.x11 }} {% endif %}{% if settings.environment_file %}-B " .. moduleDir .. "/{{ settings.environment_file }}:/.singularity.d/env/{{ settings.environment_file }}{% endif %} {% if settings.bindpaths %}-B {{ settings.bindpaths }}{% endif %} " .. containerPath
