diff --git a/lib/spack/spack/paths.py b/lib/spack/spack/paths.py
index fc617342e8..c4f7b4277d 100644
--- a/lib/spack/spack/paths.py
+++ b/lib/spack/spack/paths.py
@@ -120,7 +120,12 @@ def _get_user_cache_path():
 
 # User configuration and caches in $HOME/.spack
 def _get_user_config_path():
-    return os.path.expanduser(os.getenv("SPACK_USER_CONFIG_PATH") or "~%s.spack" % os.sep)
+    _mysoftware_path = os.getenv("MYSOFTWARE")
+    if _mysoftware_path is None:
+        user_path =  ("~%s.spack" % os.sep)
+    else:
+        user_path = f"{_mysoftware_path}{os.sep}setonix{os.sep}.spack_user_config"
+    return os.path.expanduser(user_path)
 
 
 # Configuration in /etc/spack on the system
